pynndescent/rp_trees.py

```diff
@@ -129,7 +129,7 @@ def angular_random_projection_split(data, indices, rng_state):
     return indices_left, indices_right, hyperplane_vector, None
 
 
-@numba.njit(fastmath=True, nogil=True, parallel=True)
```

Comment:
`nogil=True` is needed for some functions for threaded mode. Not sure it is for this one, but there's no harm in adding it in general.

Reply:
As a general rule it doesn't hurt if you can get away with it.